model.py

```diff
@@ -0,0 +1,391 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Thu May  7 19:32:15 2020
+
+@author: minh
+"""
+
+"""First model for the sentiment analysis problem.
+This file can be used as a model for all models.
+"""
+
+import argparse
+import logging
+
+import numpy as np
+from sklearn.model_selection import train_test_split
+
+import utils
+
+import pandas as pd
+
+import time
+import spacy
+import random
+from pathlib import Path
+import torch 
+import torch.nn as nn
+import torch.nn.functional as F
+import torch.optim as optim
+from torchtext import data 
+import torchtext
+from nltk.tokenize.treebank import TreebankWordDetokenizer
+from bs4 import BeautifulSoup
+import re
+import os
+
+# Global variables
+# ------------------------------------------
+
+MODEL_NAME = "model_lstm"
+MODEL_PATH = "{}{}.pt".format(utils.MODELS_FOLDER, MODEL_NAME)
+device = None
+
+
+class LSTM(nn.Module):
+    def __init__(self, vocab_size, embedding_dim, hidden_dim, output_dim, n_layers, 
+                 bidirectional, dropout, pad_idx):
+        """Init function.
+
+        vocab_size - vocabulary size
+        embedding_dim - size of the dense word vectors
+        hidden_dim - size of the hidden states
+        output_dim - number of classes
+        n_layers - number of multi-layer RNN
+        bidirectional - boolean - use both directions of LSTM
+        dropout - dropout probability
+        pad_idx -  string representing the pad token
+        """
+        
+        super().__init__()
+
+        # Feed in the embedding layer
+        self.embedding = nn.Embedding(vocab_size, embedding_dim, padding_idx = pad_idx)
+
+        # LSTM layer
+        self.encoder = nn.LSTM(embedding_dim, 
+                               hidden_dim, 
+                               num_layers=n_layers,
+                               bidirectional=bidirectional,
+                               dropout=dropout)
+        
+        # Fully-connected layer
+        self.predictor = nn.Linear(hidden_dim*2, output_dim)
+
+        # Initialize dropout layer for regularization
+        self.dropout = nn.Dropout(dropout)
+      
+    def forward(self, x, x_lengths):
+        """Forward function.
+        Args:
+            x (tensor): tensor of strings, shape (N,) with N the batch size
+            s (tensor): tensor of scalars, shape (N,) with N the batch size
+        """
+        embedded = self.dropout(self.embedding(x))    
+
+        packed_embedded = nn.utils.rnn.pack_padded_sequence(embedded, x_lengths) #pack the embeddings
```

Comment:
For the comment, you need a space after the "#" and you need to begin your comment by a capital letter. I will not mark all of them in the code, but same commentary.